kernel/src/listed_log_files.rs

```diff
@@ -295,17 +302,25 @@ impl ListedLogFiles {
                     .find(|(num_parts, part_files)| part_files.len() == *num_parts as usize)
                 {
                     checkpoint_parts = complete_checkpoint;
+                    // Save the latest commit file before clearing ascending_commit_files
+                    latest_commit_file = ascending_commit_files.last().cloned();
```

Comment:
We can take ownership since we're anyway clearing it
```suggestion
                    latest_commit_file = ascending_commit_files.pop();
```

Reply:
Makes sense!